core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/async/ByteArrayAsyncRequestBody.java

```diff
@@ -30,8 +30,11 @@
  * @see AsyncRequestBody#fromBytes(byte[])
  * @see AsyncRequestBody#fromByteBuffer(ByteBuffer)
  * @see AsyncRequestBody#fromString(String)
+ *
+ * @deprecated by {@link ByteBuffersAsyncRequestBody#from}.
  */
 @SdkInternalApi
+@Deprecated
 public final class ByteArrayAsyncRequestBody implements AsyncRequestBody {
```

Comment:
If we no longer use this at all, we can delete it. Internal APIs don't need to be left for backwards-compatibility.